Mathlib/Algebra/Ring/Semireal/Defs.lean

```diff
@@ -0,0 +1,114 @@
+/-
+Copyright (c) 2024 Florent Schaffhauser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Florent Schaffhauser
+-/
+import Mathlib.Algebra.Ring.Defs
+import Mathlib.Algebra.Order.Field.Defs
+
+/-!
+# Semireal rings
+
+A semireal ring is a non-trivial commutative ring (with unit) in which `-1` is *not* a sum of
+squares. Note that `-1` does not make sense in a semiring.
+
+For instance, linearly ordered fields are semireal, because sums of squares are positive and `-1` is
+not. More generally, linearly ordered semirings in which `a ≤ b → ∃ c, a + c = b` holds, are
+semireal.
+
+## Main declaration
+
+- `isSemireal`: the predicate asserting that a commutative ring `R` is semireal.
+
+## Auxiliary declarations
+
+- The predicate `isSumSq : R → Prop`: an inducitve predicate used to define sums of squares in a
+semiring `R`.
+- The type `SumSqIn R`: a subtype of `R` whose terms are the sums of squares in `R`.
+
+## Auxiliary theorems
+
+- `SumSq_nonneg`: sums of squares are non-negative.
+- `isSumSq_sum`: if `S₁` and `S₂` are sums of squares in a semiring `R`, then so is `S₁ + S₂`.
+
+## References
+
+- *An introduction to real algebra*, by T.Y. Lam. Rocky Mountain J. Math. 14(4): 767-814 (1984). [DOI: 10.1216/RMJ-1984-14-4-767](https://projecteuclid.org/journals/rocky-mountain-journal-of-mathematics/volume-14/issue-4/An-introduction-to-real-algebra/10.1216/RMJ-1984-14-4-767.full).
```

Comment:
You can add this to the bibtex file (you have to put the item in alphabetical order by hand I think), and cite it as [foo]

Reply:
I added it to the references.bib file, but I am not sure how to make the citation work, I'll try!